File: packages/web/src/types/quiz.ts
export interface Quiz {
  id: number
  section: string
  title: string
  question: string
  code?: string | undefined
  options: string[]
  correctAnswerIndex: number
  explanation: string
  source: string
}

export interface QuizAnswer {
  quizId: number
  selectedIndex: number
  correct: boolean
}

export type SessionStatus = 'in_progress' | 'completed'

export interface QuizSession {
  id: string
  sectionFilter: string | null
  total: number
  currentIndex: number
  quizIds: number[]
  answers: QuizAnswer[]
  startedAt: string
  completedAt: string | null
  status: SessionStatus
}

export interface HistoryRecord {
  id: string
  sectionFilter: string | null
  total: number
  correct: number
  startedAt: string
  completedAt: string
  answers: QuizAnswer[]
}

export interface SectionSummary {
  section: string
  count: number
}

File: packages/mobile/lib/layers/domain/entity/quiz.dart
import 'package:equatable/equatable.dart';

class Quiz extends Equatable {
  const Quiz({
    required this.id,
    required this.section,
    required this.title,
    required this.question,
    required this.options,
    required this.correctAnswerIndex,
    required this.explanation,
    required this.source,
    this.code,
  });

  final int id;
  final String section;
  final String title;
  final String question;
  final List<String> options;
  final int correctAnswerIndex;
  final String explanation;
  final String source;
  final String? code;

  String get correctAnswer => options[correctAnswerIndex];

  @override
  List<Object?> get props => [
        id,
        section,
        title,
        question,
        options,
        correctAnswerIndex,
        explanation,
        source,
        code,
      ];
}
